### ISSUE TITLE
Test the dropdown (select) function for accessibility

### ISSUE THREAD

**Issue by cjrace** [dfe-analytical-services/shinyGovstyle#61]
We had a recent accessibility audit at DfE that highlighted a few issues with standard shiny dropdown selectors. We should review whether the select function in this package would solve those issues by testing the test / template app ourselves. I believe it should, though not all GDS components are fully accessible, so this is worth checking specifically as a part of the upgrade to v5+ of the frontend system.

Examples of issues with other dropdowns we should test:
- Having a visual label marked up
- Having a non-visual label marked appropriately
- Appropriate contrast on focus

I have specific examples I can share on this to have a look at from our accessibility audit.

**Comment by sarahmwong** [comment]
hey @cjrace, please could you share the audit examples on this issue thread? Currently can tab through fine, but just want to check what we mean by label markup!

**Comment by sarahmwong** [comment]
Linking wider issue here https://github.com/alphagov/govuk-design-system-backlog/issues/60 - towards the bottom spotting issues with the contrast of the yellow highlight against white bg, plus usability issues on this vs filters

**Comment by cjrace** [comment]
Yeah sure @sarahmwong ! I can't share the report directly, but examples of the two main issues were:

1. Focus colour
1.4.1 Use of Color (Level A)

We need to have a 3:1 contrast on the focus colour, e.g. 

![Image](https://github.com/user-attachments/assets/11088935-0727-4c6a-90b7-245e49e9f0fb)

2. Label
1.3.1 Info and Relationships (Level A)
3.3.2 Labels or Instructions (Level A)
4.1.2 Name, Role, Value (Level A)

This was where someone had a select component and then floating text above like so 

![Image](https://github.com/user-attachments/assets/46c44581-eae3-4fbe-8aad-a887f5c01947)

Here the text needs to be an argument within the select function and set as the label, using the <label .. for="XXX"></label> HTML markup, the XXX being the id for the input. Biggest issues here are screen readers having no context for the select component and then also voice activation users trying to focus on the select component using the label but being unable to.

We do have the label argument in select_Input, but may need to update the HTML so the for= argument is set (maybe just needs `for=inputId` from a quick glance? Though worth testing anyway)


---

Multi-select components were massively problematic too, though I think that's a separate issue entirely.

Also, slightly related, worth keeping an eye on our DfE template as we have https://github.com/dfe-analytical-services/shiny-template/issues/100, where we're going to make examples of `shiny::selectInput()` but with GOV styling applied on top to see how that works, could be an additional option in future as it opens up the ability for server side preparing of long drop down lists

**Comment by sarahmwong** [comment]
Thanks Cam - having a play around and the colour contrast looks fine for focus on the select component here. Looks a bit different on the mac but this has been raised in the central issue of gov.uk design for this component. 

On the label - would we want the label above the select component to read "Sort by recently updated" with "recently updated" automatically updating when the user selects one of the dropdowns underneath? Or is it more that we'd want a hidden tag underneath the sorter to read "sort by recently updated" but have the visual label still just reading "sort by"? Just wanted to double check before going ahead and tweaking!

**Comment by cjrace** [comment]
I think the label itself should be just the same as the visual 'sort by' - I've not seen it update reactively elsewhere, I think that might be distracting / unnecessary extra processing for the app to do?

I'm assuming when using a screen reader it will automatically pick up and announce the selected value. So the main thing is just that the label 'sort by' is linked to the select component itself - looking at the HTML on the showcase app I don't think it is so we'll need to add in the `for=` argument

![Image](https://github.com/user-attachments/assets/0a93e19b-55b9-451d-8f75-c67b48f56a79)
